### PR TITLE
storeapi: handle 5xx error codes for all store endpoints

### PR DESCRIPTION
**Pull request by kyrofa**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

Currently, the snapcraft CLI handles a 500 response when releasing a snap. This PR fixes [LP: #1768370](https://bugs.launchpad.net/snapcraft/+bug/1768370) by making this more robust, and nicely handling 5xx codes for all endpoints.

### CODE REVIEW

File: snapcraft/storeapi/errors.py [hunk 1]
```diff
@@ -357,6 +357,28 @@ def __init__(self, result):
         super().__init__()
 
 
+class StoreInternalError(StoreError):
+
+    fmt = '{what}: {why}\n{action}'
+
+    def __init__(self, response):
+        what = 'The store encountered an internal error'
+        why = None
```
Comment (sergiusens) [inline snapcraft/storeapi/errors.py]: There will never be a case (and if there is, wee shouldn't show) any scenario where we have json for a 500. We should also leave this as a fixed message error, raised when we do not find json in a response.

Comment (kyrofa) [reply snapcraft/storeapi/errors.py]: Alright, I've updated the PR to no longer use the error list.

File: snapcraft/storeapi/errors.py [hunk 2]
```diff
@@ -369,6 +370,22 @@ def __init__(self, result):
...
 
 
+class StoreServerError(StoreError):
+
+    fmt = '{what}: {error_text} (code {error_code}).\n{action}'
+
+    def __init__(self, response):
+        what = 'The store encountered a server error'
+        error_text = responses[response.status_code].lower()
+        error_code = response.status_code
```
Comment (sparkiegeek) [inline snapcraft/storeapi/errors.py]: if you switched these lines around, you could use `error_code` to lookup `responses`

File: snapcraft/storeapi/errors.py [hunk 3]
```diff
@@ -369,6 +370,22 @@ def __init__(self, result):
         super().__init__()
 
 
+class StoreServerError(StoreError):
+
+    fmt = '{what}: {error_text} (code {error_code}).\n{action}'
+
+    def __init__(self, response):
+        what = 'The store encountered a server error'
```
Comment (evandandrea) [inline snapcraft/storeapi/errors.py]: “The Snap Store encountered an error while processing your request:”
…
“The operational status of the Snap Store can be checked at https://status.snapcraft.io”

Let’s be consistent. Snap Store is always capitalised.

Comment (kyrofa) [reply snapcraft/storeapi/errors.py]: Alright, done.

**Pull request merged**